message/inbound_msg_builder.go

```diff
@@ -242,17 +246,19 @@ func InboundChits(
 	chainID ids.ID,
 	requestID uint32,
 	preferredID ids.ID,
+	preferredIDAtHeight ids.ID,
 	acceptedID ids.ID,
 	nodeID ids.NodeID,
 ) InboundMessage {
 	return &inboundMessage{
 		nodeID: nodeID,
 		op:     ChitsOp,
 		message: &p2p.Chits{
-			ChainId:     chainID[:],
-			RequestId:   requestID,
-			PreferredId: preferredID[:],
-			AcceptedId:  acceptedID[:],
+			ChainId:             chainID[:],
+			RequestId:           requestID,
+			PreferredId:         preferredID[:],
+			PreferredIdAtHeight: preferredIDAtHeight[:],
```

Comment:
This also allows us to apply votes on a failed chit, right? Or am I too optimistic?

Reply:
We use the `AcceptedID` for failed chits. I don't think it would be safe to re-use the `PreferredIDAtHeight` in the same way unless we included an additional boolean (to say if `PreferredIDAtHeight` was accepted).